src/Raven.Server/Config/Categories/MemoryConfiguration.cs

```diff
@@ -45,5 +49,21 @@ public MemoryConfiguration()
         [DefaultValue(DefaultValueSetInConstructor)]
         [ConfigurationEntry("Memory.UseTotalDirtyMemInsteadOfMemUsage", ConfigurationEntryScope.ServerWideOnly)]
         public bool UseTotalDirtyMemInsteadOfMemUsage { get; set; }
+
+        [Description("EXPERT: Whether the high temporary dirty memory check is enabled. Default: true if the system has more than 2GB RAM")]
+        [DefaultValue(DefaultValueSetInConstructor)]
+        [ConfigurationEntry("Memory.EnableHighTemporaryDirtyMemoryUse", ConfigurationEntryScope.ServerWideOnly)]
+        public bool EnableHighTemporaryDirtyMemoryUse { get; set; }
+
+        [Description("Threshold percentage of memory for activating 'High Dirty Memory' mechanism (server will return 'Service Unavailable' for writes when scratch files dirty memory exeeds this threshold). Default: 25%")]
+        [DefaultValue(0.25d)]
+        [ConfigurationEntry("Memory.PercentageOfTemporaryDirtyMemoryAllowed", ConfigurationEntryScope.ServerWideOnly)]
+        public double PercentageOfTemporaryDirtyMemoryAllowed { get; set; }
+
+        [Description("Period in seconds between 'High Dirty Memory' checks. Default: 30 Seconds")]
+        [DefaultValue(30)]
+        [TimeUnit(TimeUnit.Seconds)]
+        [ConfigurationEntry("Memory.TemporaryDirtyMemoryChecksPeriod", ConfigurationEntryScope.ServerWideOnly)]
```

Comment:
The config needs to have unit in the name: `TemporaryDirtyMemoryChecksPeriodInSec`
Property does not have to.